src/v/cluster/scheduling/leader_balancer_random.h

```diff
@@ -156,13 +157,20 @@ class random_hill_climbing_strategy final : public leader_balancer_strategy {
                 }
             }
 
-            auto balancing_diff = _etdc.evaluate(reassignment)
-                                  + _eslc.evaluate(reassignment);
-            if (balancing_diff <= error_jitter) {
+            auto shard_load_diff = _etdc.evaluate(reassignment)
+                                   + _eslc.evaluate(reassignment);
+            if (shard_load_diff < -error_jitter) {
                 continue;
+            } else if (shard_load_diff > error_jitter) {
+                return reassignment_opt;
             }
 
-            return reassignment_opt;
+            auto node_load_diff = _enlc.evaluate(reassignment);
+            if (node_load_diff < -error_jitter) {
+                continue;
```

Comment:
This check doesn't make a difference, as we anyway continue the loop if the one below does not hold. Is it for the future when we evaluate more constraints of even lower priority below?

Reply:
Yeah, it is just for symmetry. Although I guess one more constraint and I'm going to put them in a vector :)